### PR TITLE
pr split seven - add 10 files

### PR DESCRIPTION
**Pull request by michaelkernaghan**
Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have added unit tests
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: integration-tests/contract-no-annotations-call-by-index.spec.ts [hunk 1]
```diff
@@ -3,13 +3,13 @@ import { noAnnotCode, noAnnotInit } from "./data/token_without_annotation";
...
   const Tezos = lib;
-  describe(`Test contract with no annotations calling methods by index using: ${rpc}`, () => {
+  describe(`Test call to an entrypoint with no annotations calling methods by index through contract api using: ${rpc}`, () => {
 
     beforeEach(async (done) => {
       await setup()
       done()
     })
-    it('Test contract with no annotations for methods', async (done) => {
+    it('Verify contract.originate for a contract with no annotations for method', async (done) => {
```
Comment (roxaneletourneau) [inline integration-tests/contract-no-annotations-call-by-index.spec.ts:12]: Suggestion: Verify contract.originate for a contract without annotation and call entry points

File: integration-tests/contract-originate-contract-unpair.spec.ts [hunk 2]
```diff
@@ -2,16 +2,16 @@ import { CONFIGS } from './config';
 import { miStr, miObject } from './data/contractWithUnpair';
 
 CONFIGS().forEach(({ lib, rpc, setup }) => {
-    const Tezos = lib;    
+    const Tezos = lib;
     const test = require('jest-retries');
```
Comment (roxaneletourneau) [inline integration-tests/contract-originate-contract-unpair.spec.ts:6]: Remove jest-retries?

File: integration-tests/contract-originate-contract-with-sapling.spec.ts [hunk 3]
```diff
@@ -1,29 +1,61 @@
...
 
-  describe(`Test origination of contracts with sapling using: ${rpc}`, () => {
+  describe(`Test contract origination with sapling through contract api using: ${rpc}`, () => {
     beforeEach(async (done) => {
       await setup();
       done();
     });
 
-    test('Originates a Sapling Double contract', async (done) => {
+    ithacanet('Originates a Sapling Double contract', async (done) => {
```
Comment (roxaneletourneau) [inline integration-tests/contract-originate-contract-with-sapling.spec.ts:33]: ithacanet should be removed

File: integration-tests/contract-originate-contract-with-single-sapling-state.spec.ts [hunk 4]
```diff
@@ -1,25 +1,39 @@
...
+  const jakartanetAndMondaynet = protocol === Protocols.ProtoALpha || protocol === Protocols.PtJakart2 ? test: test.skip;
 
   describe(`Test origination of contracts with sapling using: ${rpc}`, () => {
     beforeEach(async (done) => {
       await setup();
       done();
     });
 
-    test('Originates a contract with a single sapling state in its storage', async (done) => {
+    ithacanet('Originates a contract with a single sapling state in its storage for Ithaca', async (done) => {
```
Comment (roxaneletourneau) [inline integration-tests/contract-originate-contract-with-single-sapling-state.spec.ts:17]: ithacanet should be removed

File: integration-tests/contract-originate-invalid-string.spec.ts [hunk 5]
```diff
@@ -25,7 +26,11 @@ CONFIGS().forEach(({ lib, rpc, setup }) => {
           init: `"Copyright ©"`
         })
       } catch (ex) {
```
Comment (roxaneletourneau) [inline integration-tests/contract-originate-invalid-string.spec.ts:28]: Should be :
```
catch (ex) {
          expect(ex).toEqual(expect.objectContaining({ message: expect.stringContaining('non_printable_character') }))
      }
```

without distinction for protocol

File: integration-tests/contract-originate-token-contract.spec.ts [hunk 6]
```diff
@@ -7,14 +7,14 @@ CONFIGS().forEach(({ lib, rpc, setup }) => {
   const test = require('jest-retries');
```
Comment (roxaneletourneau) [inline integration-tests/contract-originate-token-contract.spec.ts:7]: Remove jest-retrie ?

File: integration-tests/contract-originate-invalid-string.spec.ts [hunk 7]
```diff
@@ -1,16 +1,17 @@
+import { Protocols } from "@taquito/taquito";
 import { CONFIGS } from "./config";
 
-CONFIGS().forEach(({ lib, rpc, setup }) => {
+CONFIGS().forEach(({ lib, rpc, setup, protocol }) => {
   const Tezos = lib;
   const test = require('jest-retries');
```
Comment (roxaneletourneau) [inline integration-tests/contract-originate-invalid-string.spec.ts:6]: remove jest-retrie?

**Review by roxaneletourneau — CHANGES_REQUESTED** [review]

File: integration-tests/contract-no-annotations-call-by-index.spec.ts (hunk 1 above)
Comment (michaelkernaghan) [reply integration-tests/contract-no-annotations-call-by-index.spec.ts:12]: changed to suggestion

File: integration-tests/contract-originate-contract-unpair.spec.ts (hunk 2 above)
Comment (michaelkernaghan) [reply integration-tests/contract-originate-contract-unpair.spec.ts:6]: removed

File: integration-tests/contract-originate-contract-with-single-sapling-state.spec.ts (hunk 4 above)
Comment (michaelkernaghan) [reply integration-tests/contract-originate-contract-with-single-sapling-state.spec.ts:17]: changed all ithacanet to kathmandunet

File: integration-tests/contract-originate-contract-with-sapling.spec.ts (hunk 3 above)
Comment (michaelkernaghan) [reply integration-tests/contract-originate-contract-with-sapling.spec.ts:33]: ithacanet removed and Jakarta/mondaynet set to just test.

File: integration-tests/contract-originate-invalid-string.spec.ts (hunk 5 above)
Comment (michaelkernaghan) [reply integration-tests/contract-originate-invalid-string.spec.ts:28]: protocol removed

File: integration-tests/contract-originate-token-contract.spec.ts (hunk 6 above)
Comment (michaelkernaghan) [reply integration-tests/contract-originate-token-contract.spec.ts:7]: removed

File: integration-tests/contract-originate-invalid-string.spec.ts (hunk 7 above)
Comment (michaelkernaghan) [reply integration-tests/contract-originate-invalid-string.spec.ts]: Retries were removed.

**Pull request merged**